### PR TITLE
Kl 464 add new hd account

### PR DESCRIPTION
**Pull request by yarmoshevych**
-account adding flow realised
-verifying is based on next criteria: after adding, test finds all account names inside account-tabs (Render Item). Writes it to array. Then verifying that array includes 'Account 2' expression

### CODE REVIEW

File: libs/ui/src/components/search-panel/search-panel.tsx [hunk 1]
```diff
@@ -13,10 +14,11 @@ import { TextInput } from '../text-input/text-input';
 import { TouchableIcon } from '../touchable-icon/touchable-icon';
 
 import { styles } from './search-panel.styles';
+import { SearchPanelTestIDs } from './search-panel.test-ids';
 
 const SEARCH_FIELD = 'search';
 
-interface Props {
+interface Props extends TestIDProps {
```
Comment (unixvb) [inline libs/ui/src/components/search-panel/search-panel.tsx]: seems like you don't need `TestIDProps` here

File: libs/ui/src/modals/screens/accounts-selector/components/accounts-list.tsx [hunk 2]
```diff
@@ -47,6 +49,7 @@ export const AccountsList: FC<Props> = ({
 
     return (
       <ModalRenderItem
+        testID={testID}
```
Comment (unixvb) [inline libs/ui/src/modals/screens/accounts-selector/components/accounts-list.tsx]: change props order: `testID` should be the last props, as in other places

File: apps/extension-e2e/src/step-definitions/account-selector.steps.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Given } from '@cucumber/cucumber';
+import { expect } from 'chai';
+
+import { Pages } from '../page-objects';
+
+Given(/I see Account 2 on the AccountsSelector page/, async () => {
+  const accountSelectorContainer = await Pages.AccountsSelector.getAccountsSelectorContainer('Account 2');
```
Comment (unixvb) [inline apps/extension-e2e/src/step-definitions/account-selector.steps.ts]: get `Account 2` as a parameter from step definition

File: apps/extension-e2e/src/page-objects/accounts-selector.page.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  async getAccountsSelectorContainer(name: string) {
+    let result = false;
+    const textElementHandler = await findElements(AccountTabsTestIDs.AccountsNames);
+    const textContents = textElementHandler.map(elementHandle =>
+      elementHandle.evaluate(element => element.textContent)
+    );
+    const namesArray = await Promise.all(textContents);
+    result = namesArray.includes(name);
+    console.log(namesArray, 'name-', name, 'result', result);
```
Comment (unixvb) [inline apps/extension-e2e/src/page-objects/accounts-selector.page.ts]: remove `console.log`

File: apps/extension-e2e/src/page-objects/accounts-selector.page.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+    return result;
```
Comment (unixvb) [inline apps/extension-e2e/src/page-objects/accounts-selector.page.ts]: the method name is `getAccountsSelectorContainer` but you are returning a boolean flag if `AccountSelectorContainer` with a specific name exists.   
return a real `AccountSelectorContainer` element or rename this function.

File: apps/extension-e2e/src/page-objects/accounts-selector.page.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  async isAccountNameDisplayed(name: string) {
+    let result = false;
+    const textElementHandler = await findElements(AccountTabsTestIDs.AccountsNames);
+    const textContents = textElementHandler.map(elementHandle =>
+      elementHandle.evaluate(element => element.textContent)
+    );
+    const namesArray = await Promise.all(textContents);
+    result = namesArray.includes(name);
+
+    return result;
```
Comment (unixvb) [inline apps/extension-e2e/src/page-objects/accounts-selector.page.ts]: just `return namesArray.includes(name);` you don't need `result` variable

File: apps/extension-e2e/src/step-definitions/account-selector.steps.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { Given } from '@cucumber/cucumber';
+import { expect } from 'chai';
+
+import { Pages } from '../page-objects';
+
+Given(/I see (.*) on the AccountsSelector page/, async (accountName: string) => {
+  const accountSelectorContainer = await Pages.AccountsSelector.isAccountNameDisplayed(accountName);
```
Comment (unixvb) [inline apps/extension-e2e/src/step-definitions/account-selector.steps.ts]: rename variable `accountSelectorContainer`

**Review by unixvb — APPROVED** [review]

**Review by vladvn1905 — APPROVED** [review]

**Pull request merged**